explorer/files/ncpi-catalog/update-bdc-source.ts

```diff
@@ -0,0 +1,60 @@
+import fetch from "node-fetch";
+import { parseContentRows, readFile } from "../../app/utils/tsvParser";
+import {
+  SOURCE_CATEGORY_KEY,
+  SOURCE_FIELD_KEY,
+  SOURCE_FIELD_TYPE,
+} from "./constants";
+import { NCPIPlatformStudy } from "./entities";
+import {
+  addNCPIHeader,
+  mergeSourceStudies,
+  replaceTsv,
+  reportStudyResults,
+  sourcePath,
+} from "./ncpi-update-utils";
+
+const urlBDC =
+  "https://gen3.biodatacatalyst.nhlbi.nih.gov/mds/metadata?_guid_type=discovery_metadata&data=false&limit=500";
+
+type BDCElement = string;
+type BDCResponse = BDCElement[];
+
+export async function updateBdcSource(sourcePath: string): Promise<void> {
+  // Extract the current studies from the source tsv
```

Comment:
Let's make a function so that we don't need to repeat lines 24-37 for each of the `update-x-source.ts` files.  Add the function to `explorer/files/ncpi-catalog/ncpi-update-utils.ts`. The goal would be to return `sourceStudies` and `sourceIds`.